pyglmnet/pyglmnet.py

```diff
@@ -678,47 +688,100 @@ def fit_predict(self, X, y):
-        method : str
-            One of 'pseudo_R2' or 'deviance'
+
+        Returns
+        -------
+        score: array | float
+            array when score is called by a list of estimators `glm.score()`
+            float when score is called by a sliced estimator `glm[0].score()`
+
+            Note that if you want compatibility with sciki-learn's
+            pipeline, cross_val_score, or GridSearchCV then you should
+            only pass sliced estimators:
+
+            from sklearn.grid_search import GridSearchCV
+            from sklearn.cross_validation import cross_val_score
+            grid = GridSearchCV(glm[0])
+            grid = cross_val_score(glm[0], X, y, cv=10)
         """
-        y = y.ravel()
-        if self.distr != 'multinomial':
-            yhat = yhat.ravel()
 
-        L1 = utils.log_likelihood(y, yhat, self.distr)
-        if self.distr in ['poisson', 'poissonexp']:
-            LS = utils.log_likelihood(y, y, self.distr)
-        else:
-            LS = 0
-
-        if method == 'deviance':
-            score = -2 * (L1 - LS)
-        elif method == 'pseudo_R2':
-            L0 = utils.log_likelihood(y, ynull, self.distr)
-            if self.distr in ['poisson', 'poissonexp']:
-                score = 1 - (LS - L1) / (LS - L0)
+        # If the model has not been fit it cannot be scored
+        if self.ynull_ is None:
+            raise ValueError('Model must be fit before prediction can \
+                              be scored')
+
+        yhat = self.predict(X)
+
+        if isinstance(self.fit_, dict):
+            # Compute scalar score for a single model fit
+            # (corresponding to a reg_lambda)
+            y = y.ravel()
+
+            if self.distr != 'multinomial':
+                yhat = yhat.ravel()
+
+            L1 = utils.log_likelihood(y, yhat, self.distr)
+            if self.distr in ['softplus', 'poisson']:
+                LS = utils.log_likelihood(y, y, self.distr)
+            else:
+                LS = 0
+
+            if self.score_metric == 'deviance':
+                score = -2 * (L1 - LS)
+            elif self.score_metric == 'pseudo_R2':
+                L0 = utils.log_likelihood(y, self.ynull_, self.distr)
+                if self.distr in ['softplus', 'poisson']:
+                    score = 1 - (LS - L1) / (LS - L0)
+                else:
+                    score = 1 - L1 / L0
+
+        elif isinstance(self.fit_, list):
+            # Compute array of scores for each model fit
+            # (corresponding to a reg_lambda)
+            y = y.ravel()
+            score = list()
+
+            if self.distr in ['softplus', 'poisson']:
+                LS = utils.log_likelihood(y, y, self.distr)
             else:
-                score = 1 - L1 / L0
+                LS = 0
+            if(self.score_metric == 'pseudo_R2'):
+                L0 = utils.log_likelihood(y, self.ynull_, self.distr)
+
+            for idx in range(yhat.shape[0]):
```

Comment:
Yes, sorry. Once I merged the previous PR there were merge conflicts and I had to merge manually and push to master.